### ISSUE TITLE
eli5 privacy/security doubt

### ISSUE THREAD

**Issue by fusionneur** [zerodytrash/Simple-YouTube-Age-Restriction-Bypass#118]
I took a quick look into the code. Can you please eli5 how does the script uses the cookies and then the proxies as I have some doubts whether there any privacy/security concern of using personal google account and this script (cookie leak, can the account be compromised?). 

*** This is not an issue, but maybe it will help others better understand not only how the script works but also to clear any doubts whther the script could be or not malicious in any way. Thank you !

**Comment by zerodytrash** [comment]
There are three different strategies to unlock the video. Depending on the type of video, this script communicates directly with the YouTube API (first two strategies) or with the account proxy (last strategy, used as fallback). The second strategy, which communicates directly with the YouTube API, does this on behalf of the logged-in user. For this purpose, an `Authorization` header must be generated, which is composed from the session cookie (`SAPISID`). Since this header is only sent to YouTube's own API, there is no security risk. The account proxy does not receive cookies or private information from the logged in user. The account proxy server uses its own session from an age-verified account stored in the backend. I hope this clears up your doubts.

**Comment by fusionneur** [comment]
Thank you for the good explanation!
